src/pages/resultsView/plots/PlotsTab.tsx

```diff
@@ -186,11 +190,15 @@ export default class PlotsTab extends React.Component<IPlotsTabProps,{}> {
                 this._dataType = t;
             },
             get dataSourceId() {
+                if (this.entrezGeneId === undefined) {
```

Comment:
this could all use some commenting to explain relatioship between gene and datasourceId.    in general we need to preserve domain knowledge better in comments.

Reply:
good point